server/src/main/java/org/elasticsearch/action/admin/indices/sampling/SamplingMetadata.java

```diff
@@ -0,0 +1,181 @@
+        return new SamplingMetadata(indexToSamplingConfigs.stream().collect(Collectors.toMap(Tuple::v1, Tuple::v2)));
+    });
+
+    static {
+        PARSER.declareNamedObjects(
+            constructorArg(),
+            (p, c, index) -> Tuple.tuple(index, SamplingConfiguration.fromXContent(p)),
+            INDEX_SAMPLING_CONFIG_MAP_PARSE_FIELD
+        );
+    }
+
+    public SamplingMetadata(Map<String, SamplingConfiguration> indexToSamplingConfigMap) {
+        this.indexToSamplingConfigMap = new HashMap<>(indexToSamplingConfigMap);
+    }
+
+    public SamplingMetadata(StreamInput in) throws IOException {
+        this.indexToSamplingConfigMap = in.readMap(StreamInput::readString, SamplingConfiguration::new);
+    }
+
+    public Map<String, SamplingConfiguration> getIndexToSamplingConfigMap() {
+        return indexToSamplingConfigMap;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeMap(indexToSamplingConfigMap, StreamOutput::writeString, (o, v) -> v.writeTo(o));
+    }
+
+    public static SamplingMetadata fromXContent(XContentParser parser) throws IOException {
+        return PARSER.parse(parser, null);
+    }
+
+    public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+        builder.startObject();
+        {
+            builder.startObject(INDEX_SAMPLING_CONFIG_MAP_FIELD_NAME);
+            for (Map.Entry<String, SamplingConfiguration> e : indexToSamplingConfigMap.entrySet()) {
+                builder.field(e.getKey(), e.getValue());
+            }
+            builder.endObject();
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        SamplingMetadata that = (SamplingMetadata) o;
+        return Objects.equals(indexToSamplingConfigMap, that.indexToSamplingConfigMap);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(indexToSamplingConfigMap);
+    }
+
+    @Override
+    public EnumSet<Metadata.XContentContext> context() {
+        return Metadata.ALL_CONTEXTS;
+    }
+
+    @Override
+    public String getWriteableName() {
+        return TYPE;
+    }
+
+    @Override
+    public TransportVersion getMinimalSupportedVersion() {
+        return TransportVersions.SAMPLING_CONFIGURATION;
+    }
+
+    @Override
+    public Iterator<? extends ToXContent> toXContentChunked(ToXContent.Params params) {
+        return ChunkedToXContentHelper.xContentObjectFields(
+            INDEX_SAMPLING_CONFIG_MAP_PARSE_FIELD.getPreferredName(),
+            indexToSamplingConfigMap
+        );
+    }
+
+    @Override
+    public Diff<Metadata.ProjectCustom> diff(Metadata.ProjectCustom previousState) {
+        return null;
```

Comment:
I think you want `return new SamplingMetadataDiff((SamplingMetadata) previousState, this);` here